### PR TITLE
[scroll-animations-1] Scroll timelines are inactive in paged media. #8226

### PR DESCRIPTION
**Pull request by fantasai**

### CODE REVIEW

File: scroll-animations-1/Overview.bs [hunk 1]
```diff
@@ -210,6 +210,11 @@ spec: cssom-view-1; type: dfn;
 			then the [=scroll progress timeline=] is [=inactive timeline|inactive=].
 	</dl>
 
+	References to the [=root element=] propagate to the document viewport
+	(which functions as its [=scroll container=]).
+	However in [=paged media=],
+	the [=root element=]’s [=scroll progress timeline=] is [=inactive timeline|inactive=].
```
Comment (flackr) [inline scroll-animations-1/Overview.bs:216]: This implies that we have a single scroll progress timeline for anything that references the root (whether explicitly or nearest), but they do have different identities. How about something like:

However in [=paged media=],
[=scroll progress timelines=] referring to the document viewport are [=inactive timeline|inactive=]

File: scroll-animations-1/Overview.bs [hunk 2]
```diff
@@ -498,6 +503,8 @@ spec: cssom-view-1; type: dfn;
 
 	If the 0% position and 100% position coincide
 	(i.e. the denominator in the [=current time=] formula is zero),
+	or if there is no corresponding [=scroll offset=]
```
Comment (flackr) [inline scroll-animations-1/Overview.bs:506]: Doesn't no corresponding scroll offset imply this happens when there's no scrollable overflow - which is not strictly true? I was thinking we could just say the [=view progress visibility range=] is the scrollable overflow size and let the rest fall out from the math. That said, I'm okay with making timelines looking at the root scroll on printed media blanket inactive I just don't want this to apply to other cases where we have well established behaviors as the overflow becomes entirely visible, e.g. demo in https://github.com/w3c/csswg-drafts/issues/7778#issuecomment-1310360733

Comment (fantasai) [reply scroll-animations-1/Overview.bs:506]: > Doesn't no corresponding scroll offset imply this happens when there's no scrollable overflow - which is not strictly true?

I was expecting that any scroll container has to have a scroll offset, even if it's always zero.

> I was thinking we could just say the [=view progress visibility range=] is the scrollable overflow size and let the rest fall out from the math.

It becomes a little bit vague what that means for paged media, especially if the content is longer than the root element itself.

**Comment by fantasai** [comment]
@flackr OK, I posted two commits into the PR, with two possible options for wording. Lmk what you think, if you prefer the first commit or the second one. :)

**Comment by flackr** [comment]
I think I prefer the second. I'm a little torn because the general concept of "paged media" doesn't have the same constraints as "print" but they seem to be a bit conflated. I.e. An ebook reader could be considered paged (I think) but *does* actually have a changing scroll progress and you may well want to use that to update a progress indicator with a scroll driven animation.

**Comment by flackr** [comment]
> But I totally agree with you that it could make sense for a dynamic paged medium to have an active timeline. Should we open an issue on that? I think there's some other considerations, though, since a number of scroll animations are designed in a way that only makes sense on continuous media (see e.g. various NY Times articles), and if we default to applying those to dynamic paged media those devices could end up with a lot of unreadable pages. :/

I filed #8688 to discuss :-)

**Pull request merged**